### ISSUE TITLE
Merritt Maintenance Causes SWORD/Local ID issues

### ISSUE THREAD

**Issue by sfisher** [CDLUC3/mrt-doc#644]
**Describe the bug**

Around Merritt maintenances, there are occasionally problems where Merritt splits datasets submitted through Merritt-sword into new datasets because of issues with a local-id service on the Merritt side.

Merritt-Sword uses URLs like this for submissions and updates.  `http://mrtsword.cdlib.org:39001/mrtsword/edit/cdl_dryad/doi%3A10.5061%2Fdryad.4f4qrfj8s`  .  The URLs contain DOIs only and no ARKS (local ids) and are the SWORD way of submitting new versions of an existing dataset to Merritt.  When the SWORD service doesn't recognize that the local-id service isn't operating properly and is borked, then we get split datasets sometimes.

Mark and I found one the other day from Tuesday's maintenance.  The link below is another.  I hear from curators that there is probably another that isn't downloading right and is likely caused by this same issue.

https://merritt.cdlib.org/collection/search_results?utf8=%E2%9C%93&terms=4f4qrfj8s&group=cdl_dryad&commit=Go

Unfortunately, these become fairly annoying to correct since we have two datasets where there should only be one and causes 1) versioning to be incorrect so versions after the split don't download correctly and 2) we have to go resubmit everything after the split back to the original dataset to correct it and then delete the bogus extra dataset and mess with some things in our database.

Another local_id problem from Merritt-Sword around Tuesday's maintenance that users are complaining about.  Dataset submitted with the same Merritt Update URL resulted in two different ARKS.  Both have doi:10.5061/dryad.4f4qrfj8s as shown in the metadata files.

- https://merritt.cdlib.org/m/ark%253A%252F13030%252Fm5b6298m
- https://merritt.cdlib.org/m/ark%253A%252F13030%252Fm5v46kps

**To Reproduce**

???

1. Have a maintenance
2. Bork the local-id service or restart other Merritt services before the local-id service is working correctly.  I believe the errors on Tuesday were gateway errors or timeout errors or something where one of the services in the rotation was having problems.  IDK what the precise error is maybe the service was timing out or returning HTTP 502 status codes or something else.  I think Mark knows more.
3. Create a new dataset instead of updating the expected dataset.

**Expected behavior**

If there is a way to catch local-id errors in Merritt-Sword and fail submissions rather than continuing on, it would be really preferable since cleanly failed submissions are easy to fix and can simply be re-submitted when things are working correctly again. Creating new dataset submissions when local-id isn't working correctly is harder to fix and generates more manual cleanup work.

**Comment by mreyescdl** [comment]
@sfisher 
Below are the nine remaining split Dryad objects.  The local ID that Merritt knows of targets the object with the most versions.
I'm not sure if Dryad knows of the object that only has one version and no Merritt local ID.  
Ideally, we can just remove this object after verifying that the payload is contained in the other object 

Root cause of most of these was a misconfigured Inventory ALB (much like the recent split object) on 9/3/2020

ark:/13030/m5673kv1     doi:10.5061/dryad.t4b8gthxd     5 versions created 2019-10-14
ark:/13030/m56f10m5     doi:10.5061/dryad.t4b8gthxd     1 version created 2019-10-13
- Local ID points to ark:/13030/m5673kv1

ark:/13030/m50d030r     doi:10.5061/dryad.34tmpg4gj     2 versions created 2020-03-16 01:58:14
ark:/13030/m5vm9kkd     doi:10.5061/dryad.34tmpg4gj     1 version created 2020-03-16 01:58:14
- Local ID points to ark:/13030/m50d030r

ark:/13030/m5fj7vfr     doi:10.5061/dryad.2ngf1vhkw     2 versions created 2020-09-04 15:30:33
ark:/13030/m5vx5v4g     doi:10.5061/dryad.2ngf1vhkw     1 version created 2020-09-03 19:42:10
- Local ID points to ark:/13030/m5fj7vfr

ark:/13030/m5s524mq     doi:10.5061/dryad.mkkwh70xd     4 versions created 2020-09-08 07:12:10
ark:/13030/m5bw2v67     doi:10.5061/dryad.mkkwh70xd     1 version created 2020-09-03 23:21:54
- Local ID points to ark:/13030/m5s524mq

ark:/13030/m5005dr3     doi:10.5061/dryad.vx0k6djpr     5 versions created 2020-09-06 13:20:30
ark:/13030/m53c1btc     doi:10.5061/dryad.vx0k6djpr     1 version created 2020-09-04 00:49:56
- Local ID points to ark:/13030/m5005dr3

ark:/13030/m5gj4wk4     doi:10.5061/dryad.fn2z34trw     4 versions created 2020-09-04 11:39:26
ark:/13030/m5zp9jqd     doi:10.5061/dryad.fn2z34trw     1 version created 2020-09-04 01:07:56
- Local ID points to ark:/13030/m5gj4wk4

ark:/13030/m5713d1f     doi:10.5061/dryad.12jm63xw3     2 versions created 2020-09-04 12:07:27
ark:/13030/m5q58265     doi:10.5061/dryad.12jm63xw3     1 version created 2020-09-04 02:16:57
- Local ID points to ark:/13030/m5713d1f

ark:/13030/m5ms95dr     doi:10.5061/dryad.0rxwdbrxg     3 versions created 2020-09-06 08:23:27
ark:/13030/m5907gcj     doi:10.5061/dryad.0rxwdbrxg     1 version created 2020-09-04 07:30:17
- Local ID points to ark:/13030/m5ms95dr

ark:/13030/m5c58zfz5    doi:10.5061/dryad.v15dv41tq     3 versions created 2020-09-07 16:18:02
ark:/13030/m5810f85     doi:10.5061/dryad.v15dv41tq     1 version created 2020-09-04 08:23:19
- Local ID points to ark:/13030/m5c58zfz

**Comment by elopatin-uc3** [comment]
Noting associated PR from Mark: https://github.com/CDLUC3/mrt-ingest/pull/18

**Comment by mreyescdl** [comment]
@sfisher Looking at all the payloads, there is only one difference that I can see.
The object that does not contain the local ID, and thus is not active to Dryad contains a file that is missing.

ark:/13030/m5gj4wk4     doi:10.5061/dryad.fn2z34trw     4 versions created 2020-09-04 11:39:26
ark:/13030/m5zp9jqd     doi:10.5061/dryad.fn2z34trw     1 version created 2020-09-04 01:07:56
- Local ID points to ark:/13030/m5gj4wk4
**** Missing sedimentation_data.zip ****

**Comment by sfisher** [comment]
Hi @mreyescdl Thanks for putting together this list.  That's super helpful.

I went through all of them in the list and we had fixed most of them in an earlier round of fixing.  We're ignoring the single version that was missing the local ID for all of these, so those can be deleted from Merritt if you want to do that for cleanup.

I did find one that had some incorrect versioning (though the file was present and they were using an earlier version in public so downloads were not broken).  I fixed by changing the Merritt version number in our database, so if the author submits another version or two it will not break in the future.

Thanks again for doing all the work in finding these and it seems like we're in good shape now.
